packages/adapter/src/request-building/authentication.ts

```diff
@@ -74,32 +73,16 @@ export function buildParameters(options: CachedBuildRequestOptions): Authenticat
     cookies: {},
   };
 
-  const { securitySchemeSecrets } = options;
-  // Security schemes originate from 'config.json' and contain all of the authentication details
-  // but the actual secrets and values needed by the Airnode deployment will come from 'secrets.env'
-  if (!securitySchemeSecrets || isEmpty(securitySchemeSecrets)) {
+  const { credentials } = options;
+  if (!credentials) {
     return initialParameters;
   }
 
   // API security schemes also originate from 'config.json' and specify which schemes should be used
-  const apiSecuritySchemes = options.ois.apiSpecifications.components.securitySchemes;
-  if (isEmpty(apiSecuritySchemes)) {
+  const apiSecurityScheme = options.ois.apiSpecifications.components.securitySchemes[credentials.securityScheme];
+  if (!apiSecurityScheme) {
     return initialParameters;
   }
 
-  const apiSecuritySchemeNames = Object.keys(apiSecuritySchemes);
```

Comment:
Do I understand correctly that this just merged security schemes along with the secrets and we don't need to do that for all schmes, just the one that will be used?

Reply:
We'll discuss how we want to deal with this in the end but yes, for now, we have multiple scheme definitions in `ois.components.securitySchemes` and are selecting one that is to be used with `ois.credentials`.